shared/js/background/declarative-net-request.js

```diff
@@ -327,6 +331,16 @@ export async function refreshUserAllowlistRules (allowlistedDomains) {
     await updateUserAllowlistRule(normalizedAllowlistedDomains)
 }
 
+export async function flushSessionRules () {
+    chrome.declarativeNetRequest.getSessionRules().then(rules => {
```

Comment:
Nit: Might as well use `await` here, since the function is `async` anyway.

Reply:
That is handled by the caller (which is the right pattern IMHO, let the caller decide whether to await or resolve as promise)

Reply:
The caller will be returned a Promise either way, since this is an async function. The caller can wait (or not) for that Promise to resolve either way. My suggestion would just mean that the returned Promise would resolve to the result (if any) of updateSessionRules, instead of resolving to a Promise that in turn resolves to a Promise which in turn resolves to the result on updateSessionRules.

Reply:
Oops you're right, insofar I forgot to remove the `async`.  :)

I noticed we seem to have fallen a bit into the pattern of using `await` just for return values, which in turn requires the containing function itself to be async, which is a NOOP in practice and creates cognitive overhead.   `refreshUserAllowlistRules()` right above is an example of this.

Maybe this better illustrates my point?

```
async function go() {
    function a1() {
        return Promise.resolve(true)
    }

    function a2() {
        return a1()
    }

    function a3() {
        return a2()
    }

    console.log('go', a3())
    console.log('await go ', await a3())
}

async function gonuts() {
    async function a1() {
        return await Promise.resolve(true)
    }

    async function a2() {
        return await a1()
    }

    async function a3() {
        return await a2()
    }

    console.log('gonuts', a3())
    console.log('await gonuts', await a3())
}

go()
gonuts()
```
results in
```
go Promise { true }
gonuts Promise { <pending> }
await go  true
await gonuts true
```

So I think if we see patterns where we're just doing `async .... return await x()`, that should give us pause?